CasadiFunctions/createCasadi_Ligaments.m

```diff
@@ -0,0 +1,178 @@
+
+    idx_coord_single = find(sum(model_info.ligament_info.ligament_spanning_single_coord,1));
+    
+    for i=idx_coord_single
+        q_i = model_info.ligament_info.polyFit.DummySamples.q(:,i);
+        M_i = zeros(size(q_i));
+    
+        idx_lig = find(model_info.ligament_info.ligament_spanning_single_coord(:,i));
+    
+        % sort angle and moment in order of angle
+        [q_i,idx_sort] = sort(q_i);
+
+        for j=idx_lig'
+            % test for valid input force-length
+            if ~exist(model_info.ligament_info.parameters(j).stiffness,'file')
+                error(['Unable to find ' model_info.ligament_info.parameters(j).stiffness,...
+                    '. Add the function to ./ModelComponents, or select an existing function for ',...
+                    'the force-length of ligament ' model_info.ligament_info.parameters(j).ligament_name '.'])
+            end
+    
+            % evaluate force-length for dummy motion
+            f_stiffness = str2func(model_info.ligament_info.parameters(j).stiffness);
+            CSA = model_info.ligament_info.parameters(j).cross_section_area;
+            ls = model_info.ligament_info.parameters(j).slack_length;
+            l_lig = model_info.ligament_info.polyFit.DummySamples.lMT(idx_sort,j);
+            d_lig = model_info.ligament_info.polyFit.DummySamples.dM(idx_sort,j,i);
+            F_lig_j = f_stiffness(CSA,ls,l_lig);
+    
+            % ligament length and force for use in post-processing
+            f_L_lig_j = interpolant(['f_L_lig_' model_info.ligament_info.parameters(j).ligament_name],...
+                'bspline',{q_i},l_lig);
+            f_F_lig_j = interpolant(['f_F_lig_' model_info.ligament_info.parameters(j).ligament_name],...
+                'bspline',{q_i},F_lig_j);
+            
+            L_lig_pp_MX(j) = f_L_lig_j(Qs_MX(i));
+            F_lig_pp_MX(j) = f_F_lig_j(Qs_MX(i));
+
+
+            % calculate moment
+            M_lig_j = F_lig_j.*d_lig;
+            M_i = M_i + M_lig_j;
+        end
+    
+        % create look-up table
+        f_lig_single_coord = interpolant(['f_lig_single_coord_',...
+            model_info.ExtFunIO.coord_names.all{i}], 'bspline',{q_i},M_i);
+        
+        % assign to coordinate
+        M_lig_single_MX(i) = f_lig_single_coord(Qs_MX(i));
+    
+    end
+    
+    
+    %% Ligaments that cross multiple coordinates
+    idx_multi_coord = find(sum(model_info.ligament_info.ligament_spanning_multi_coord,2));
+
+    % Assemble polynomial approximation from coefficients
+    qin_SX = SX.sym('qin',model_info.ExtFunIO.jointi.nq.all,1);
+    l_lig_p_SX = SX(model_info.ligament_info.NLigament,1);
+
+    if sum(model_info.ligament_info.ligament_spanning_multi_coord,'all') > 0
+        ligament_spanning_info_m = model_info.ligament_info.ligament_spanning_multi_coord(:,:);
+        LigamentInfo_m.muscle = model_info.ligament_info.polyFit.LigamentInfo.muscle(:);
+        load nCoeffMat
```

Comment:
This is not good practice.  You should specify the names of the variables you load from .mat file
(e.g. load(myfile.mat,'varname1','varname2')

Is a full path do the datafile (instead of just in the current folder also not needed)